rel/fed1.vars.config

```diff
@@ -14,7 +14,7 @@
 {hosts, "[ \"fed1\" ]"}.
 {sm_backend, "{mnesia, []}"}.
 {auth_method, "internal"}.
-{s2s_addr, "{ {s2s_addr, \"localhost\"}, {127,0,0,1} }. { {s2s_addr, \"localhost.bis\"}, {127,0,0,1} }."}.
+{s2s_addr, "{ {s2s_addr, \"localhost\"}, {127,0,0,1} }. { {s2s_addr, \"pubsub.localhost\"}, {127,0,0,1} }. { {s2s_addr, \"muc.localhost\"}, {127,0,0,1} }. { {s2s_addr, \"localhost.bis\"}, {127,0,0,1} }."}.
```

Comment:
doesn't `{s2s_addr, "{ {s2s_addr, \"localhost\"}, {127,0,0,1} }.` cover sub-domains (`pubsub.localhost` & `muc.localhost`)?

Reply:
I believe not, for two reasons:
1. The `ejabberd_s2s_out` code looks up the full XMPP domain name in the `s2s_addr` list via `get_predefined_addresses`, which doesn't have any logic related to subdomains.  https://github.com/esl/MongooseIM/blob/master/src/ejabberd_s2s_out.erl#L1198
2. Changing this setting resolved `remote-server-not-found` in the `pubsub_s2s_SUITE` results.

Reply:
It seems that adding `{route_subdomains, s2s}.` option solves this problem. see [documentation](https://mongooseim.readthedocs.io/en/latest/Advanced-configuration/#served-hostnames) for more information.

Reply:
I don't see how it will help.  The problem is fed1 (which serves XMPP domain `fed1`) trying to contact mim1 (which serves XMPP domains `localhost` and `pubsub.localhost`).

Configuring `route_subdomains` on fed1 would change how fed1 routes messages to `subdomain.fed1`, and configuring `route_subdomains` on mim1 would change how mim1 routes messages to `subdmain.localhost`, but the problem is how fed1 routes messages to `subdomain.localhost`, which would not be affected as fed1 does not serve XMPP domain `localhost`.

Reply:
Yes, you're right. I misinterpreted it. just had a quick conversation within the team. All subdomains must be explicitly configured using DNS or s2s_addr option. That is the only right and possible way.